binding.cc

```diff
@@ -1107,6 +1197,77 @@ namespace zmq {
     BufferReference* bufref_;
   };
 
+  NAN_METHOD(Socket::Sendv) {
+    Socket* socket = GetSocket(info);
+    if (socket->state_ != STATE_READY)
+      return info.GetReturnValue().Set(Nan::False());
+
+    int events;
+    size_t events_size = sizeof(events);
+
+    int rc;
+
+    while (true) {
+      rc = zmq_getsockopt(socket->socket_, ZMQ_EVENTS, &events, &events_size);
+      if (rc != 0) {
+        if (zmq_errno()==EINTR) {
+          continue;
+        }
+        return Nan::ThrowError(ErrorMessage());
+      }
+      break;
+    }
+
+    if ((events & ZMQ_POLLOUT) == 0) {
+      return info.GetReturnValue().Set(Nan::False());
+    }
+
+    Local<Array> batch = info[0].As<Array>();
+    size_t len = batch->Length();
+
+    if (len == 0)
+      return info.GetReturnValue().Set(Nan::True());
+
+    if (len % 2 != 0)
+      return Nan::ThrowTypeError("Batch length must be even!");
+
+    for (size_t i = 0; i < len; i += 2) {
+      Local<Object> buf = batch->Get(i).As<Object>();;
+      Local<Number> flagsObj = batch->Get(i + 1).As<Number>();;
+
+      int flags = Nan::To<int>(flagsObj).FromJust();
```

Comment:
`Local<Integer> flagsObj = batch->Get(i + 1).As<Integer>();` is more indicative
alternatively `Local<Int32> flagsObj = batch->Get(i + 1).As<Int32>();`, if it can be guaranteed to only need 32 bits